Makefile

```diff
@@ -1,22 +1,127 @@
-.PHONY: help clean spec mystspec
+# Because env `SHELL` might be 
+# some extremely weird shell
+SHELL = bash
 
-spec:      ## Runs all specs
-	crystal spec
-	crystal run src/myst_cli.cr -- spec/myst/spec.mt
+# Parse makefiles named *.mk in the dir `local`
+# for local extra makefile targets if wanted.
+# It is smart to set MYSTBUILD to your myst dev-build there
+-include local/*.mk
 
-myst-spec: ## Runs just the in-language specs
-	crystal run src/myst_cli.cr -- spec/myst/spec.mt
-	
-build:     ## Builds myst into an executable
-	shards build
+# https://www.gnu.org/software/make/manual/html_node/One-Shell.html
+.ONESHELL:
 
-check:     ## Runs all crystal specs
-	crystal spec/
+# The docs says i should do this
+.SUFFIXES: .cr .mt
 
-clean:     ## Cleans (deletes) docs and executables
-	rm -rf docs
-	rm bin/*
+MYST_CLI = src/myst_cli.cr
 
-# https://gist.github.com/prwhite/8168133 "Help target"
-help:      ## Show this help.
-	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
+MYST_INTERPRETER_SPEC = spec/all_spec.cr
+MYST_IN_LANG_SPEC 		= spec/myst/spec.mt
+
+SOURCE_FILES := $(shell find src    -type f -name '*.cr')
+SPEC_FILES   := $(shell find spec   -type f -name '*.cr' -o -name '*.mt')
+STDLIB_FILES := $(shell find stdlib -type f -name '*.mt')
+ALL_FILES    := $(SPEC_FILES) $(STDLIB_FILES) $(SOURCE_FILES)
+
+INSTALL_LOCATION ?= /usr/local/bin/myst
+MYSTBUILD				 ?= myst
+
+O ?= bin
```

Comment:
Is `0` supposed to represent the "object" directory? I'm not a fan of single character names, so I'd like it if this could be expanded to something more descriptive, or at least have a comment about what it represents.

Reply:
`O` is supposed to represent 'out', i'll just change it to that, this is how crystal's makefile does it, so I just thought they knew best. But, I agree